flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -565,7 +565,20 @@ static ContainerLaunchContext createTaskExecutorContext(
 						new File(fileLocation),
 						HadoopUtils.getHadoopConfiguration(flinkConfig));
 
-				cred.writeTokenStorageToStream(dob);
+				// Filter out AMRMToken before setting the tokens to the TaskManager container context.
+				Method getAllTokensMethod = Credentials.class.getMethod("getAllTokens");
+				Credentials taskManagerCred = new Credentials();
+				final Text amRmTokenKind = new Text("YARN_AM_RM_TOKEN");
```

Comment:
Can we refactor `YARN_AM_RM_TOKEN` to `static final ` indicator of the Utils class? it will also be used in test as well.

Reply:
Good point, I will make it as a `static final` member.